examples/ui/display_and_visibility.rs

```diff
@@ -433,16 +433,21 @@ where
 }
 
 fn buttons_handler<T>(
+    mut click_events: EventReader<Click>,
+    visibility_button_query: Query<(&Target<T>, &Children)>,
     mut left_panel_query: Query<&mut <Target<T> as TargetUpdate>::TargetComponent>,
-    mut visibility_button_query: Query<(&Target<T>, &Interaction, &Children), Changed<Interaction>>,
     mut text_query: Query<&mut Text>,
 ) where
     T: Send + Sync,
     Target<T>: TargetUpdate + Component,
 {
-    for (target, interaction, children) in visibility_button_query.iter_mut() {
-        if matches!(interaction, Interaction::Pressed) {
+    for event in &mut click_events.read() {
+        if let Ok(clicked_visible_button) = visibility_button_query.get(event.0) {
+            let target = clicked_visible_button.0;
+            let children = clicked_visible_button.1;
```

Comment:
Let's unpack it right away:
```suggestion
        if let Ok((target, children)) = visibility_button_query.get(event.0) {
```